frontend/src/components/common/table/Table.tsx

```diff
@@ -52,8 +54,10 @@ export const Table = <T extends Nodes = Nodes>({
           <Tbody>
             {paginatedData.map((row, rowIndex) => (
               <Tr
-                key={rowIndex}
+                key={row.id || rowIndex}
```

Comment:
Just a thought: defaulting to the index for a common component like a table could be dangerous if we're navigating between pages. Could also be fine though depending on the circumstances